code/game/machinery/ftl_shieldgen.dm

```diff
@@ -206,3 +225,59 @@
 	if(get_dir(src, target) == in_dir)
 		return 1
 	return 0
+
+/obj/effect/ftl_shield/Initialize(var/newloc)
+	..(newloc)
+	spawn(1) // Updates neightbors after we're gone.
+		for(var/direction in GLOB.cardinals)
+			var/turf/T = get_step(src.loc, direction)
+			if(T)
+				for(var/obj/effect/ftl_shield/F in T)
+					F.update_icon()
+
+/obj/effect/ftl_shield/attack_hand(var/mob/living/user)
+	impact_effect(3) // Harmless, but still produces the 'impact' effect.
+	..()
+
+/obj/effect/ftl_shield/Bumped(atom/A)
+	..(A)
+	impact_effect(2)
+
+
+/obj/effect/ftl_shield/update_icon(var/update_neightbors = 0)
+	overlays.Cut()
+	var/list/adjacent_shields_dir = list()
```

Comment:
this should be a var on the shield and updated when a shield is moved or deleted